src/prefect/orion/api/block_documents.py

```diff
@@ -74,7 +65,7 @@ async def read_block_documents(
     return result
 
 
-@router.get("/{id}")
+@router.get("/{id:uuid}")
```

Comment:
Is this necessary to get the desired behavior?

Reply:
It is, because without that, we get 422s instead of 404s, because the FastAPI kind of assumes that if it makes it as far as a `Depends`, then the URL must be correct it's something in the body of the request that's wrong.  This is the more correct form anyway, using a [starlette path converter](https://www.starlette.io/routing/#path-parameters).

Reply:
Hmmm is 422 the correct status to return though?
Providing something other than a UUID means we can't process the request. And it's probably useful for the client to know they've provided a bad data type rather than "we couldnt find this" right?

Reply:
I believe 404 is the correct status here.  They are requesting an object that doesn't exist ("/block_documents/null")

Here's how I interpret the status codes:
400: something in the _request line_ or _header_ is incorrect
401: you aren't allowed, maybe try authenticating
403: you aren't allowed, and authenticating won't help you

Reply:
We could get this error even on a `GET` request today, which doesn't even have a body, so I'm pretty sure 422 doesn't work.